src/pages/NewsList/NewsList.tsx

```diff
@@ -102,29 +94,67 @@ export default function NewsList() {
 
 			{data && data.length > 0 ? (
 				<div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-5 min-w-96">
-					{data.map((data: NewsData) => (
+					{data.map((newsItem: NewsData) => (
 						<div
-							key={`${data.url}-${data.time_published}`}
-							className="flex flex-col h-full p-6 mb-4 bg-gray-50 rounded-lg shadow-lg"
+							key={`${newsItem.url}-${newsItem.time_published}`}
+							className="flex flex-col h-full p-6 mb-6 bg-gray-50 rounded-lg shadow-lg"
 						>
-							<h2 className="text-xl font-bold mb-2 text-gray-900">{data.title}</h2>
-							<p className="text-gray-700">{data.summary}</p>
-					{newsData.map((news: NewsData) => (
-						<div key={`${news.url}-${news.time_published}`}
-							 className="flex flex-col h-full p-6 mb-4 bg-gray-50 rounded-lg shadow-lg">
-							{news.banner_image ? (
-								<img src={news.banner_image} alt={news.title} className="mb-4 rounded-xl shadow-lg" />
-							) : (
-								<img src={placeholder} alt="article" className="mb-4 rounded-xl shadow-lg" />
-							)}
-							<h2 className="text-xl font-bold mb-2 text-gray-900">{news.title}</h2>
-							<p className="text-gray-700">{news.summary}</p>
+							<div className="relative">
+								{newsItem.banner_image ? (
+									<img
+										src={newsItem.banner_image}
+										alt={newsItem.title}
+										className="h-[180px] xl:h-[250px] mb-6 w-full object-cover rounded-xl shadow-lg"
+									/>
+								) : (
+									<img
+										src={placeholder}
+										alt={newsItem.title}
+										className="h-[180px] xl:h-[250px] mb-6 w-full object-cover rounded-xl shadow-lg"
+									/>
+								)}
+								<ul className="absolute bottom-7 right-0">
+									{newsItem.topics.slice(0, 3).map((topicData, index) => {
+										let className = 'py-1 px-3 mb-1 rounded-l-full shadow-lg text-sm opacity-85';
+
+										if (topicData.topic === 'Earnings') {
```

Comment:
I would prefer to create an object (at the top of the file), where you would match the color and name. Then you can check if the topic is in the object, then use its color and if not use a fallback.